deploy.go

```diff
@@ -225,11 +225,10 @@ func (p *deploycmd) deployFunc(c *cli.Context, appName, baseDir, funcfilePath st
 	}
 	fmt.Printf("Deploying %s to app: %s at path: %s\n", funcfile.Name, appName, funcfile.Path)
 
-	funcfile2, err := bumpIt(funcfilePath, Patch)
+	funcfile, err := bumpVersion(funcfile, Patch)
```

Comment:
I believe that this change will have unintended consequences.. I am not sure did you look at the side effects of doing this?

Reply:
there are no unintended consequences because bumpVersion only bumps the version in `funcfile` struct, it doesn't write to the file, but `bumpIt` does both which is wrong, CLI should do the following steps:
 1. bump version at `funcfile` struct only
 2. run deploy
 3. write file

So, in case of failure at step 2 func version will remain the same which is good, failed build must not change the function version.

Reply:
What happens if two succeeds, but three doesn't, that would be bad. Might be better to write file before deploying.

Reply:
That's exactly what it is right now, and this is what it makes function version grow with every unsuccessful deploy which is the wrong behaviour. If step 3 is not successful it means that someone (not the CLI) corrupted func file, at that moment we it's not our fault and we should not do something unexpected.

Reply:
The problem is that if we deploy without being 100% sure the func file is updated, then the next time you deploy, you will basically overwrite the previous version, which is bad. It's better to have your version numbers grow since that has zero chance of any side effects.

Reply:
“Next time deploy” really means that the previous deploy for 100% unsuccessful, so, obviously, broken deploy doesn’t mean new version in any possible case. So, if we try to rerun the deploy it would override built image only in one case - someone corrupted func.yml which means that function version (from func.yml) corrupted as well, so we don’t know which previous versions function had. Override is an expected behaviors. I’d like us to follow exactly the same logic that docker uses - it doesn’t declare new tag unless that was explicitly specified, so the CLI should do the same - build-bump, but not bump-no-matter-what-then-build.

Reply:
You understand the risk right?  

Why does it matter if the bump happens before the build? Because a number is skipped?  Who cares? There's zero risk that way.

Reply:
Okay, here's how i see it. We do what we were doing before (update func file with new version before starting the build), but we should definitely revert func file in case if build was faulty.